src/puppetlabs/puppetdb/import.clj

```diff
@@ -48,8 +49,25 @@
                       (utils/read-json-content tar-reader)))
       nil)))
 
+(def metadata-path
+  (.getPath (io/file utils/export-root-dir export/export-metadata-file-name)))
+
+(defn parse-metadata
+  "Parses the export metadata file to determine, e.g., what versions of the
+  commands should be used during import."
+  [tar-reader]
+  {:post [(map? %)
+          (:command_versions %)]}
+  (when-not (= (.getName (archive/next-entry tar-reader))
```

Comment:
`archive/next-entry` is side-affecting, right? If that's true, I think it would be better to have an explicit `(let [entry (archive/next-entry tar-reader)] ...)` form. That makes it a bit more clear that we're dealing with the side effect.